### PR TITLE
[SR SQL Planner Summer Camp] add showBackup analyzer

### PR DESCRIPTION
**Pull request by melt-code**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7951

## Problem Summary(Required) ：

add showBackup analyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 30 / 36 (83.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[418]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|2|50.00%|[111]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|21|25|84.00%|[20, 40, 41, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -789,6 +792,11 @@ showProcStatement
     : SHOW PROC path=string
     ;
 
+// ---------------------------------------- Backup Store Statement -----------------------------------------------------
+showBackupStatement
+    : SHOW BACKUP (FROM identifier)?
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:797]:
```suggestion
    : SHOW BACKUP ((FROM | IN) identifier)?
```

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowBackupAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.common.FeNameFormat;
+import com.starrocks.common.UserException;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.CatalogMgr;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.ast.AstVisitor;
+
+public class ShowBackupAnalyzer {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowBackupAnalyzer.java:20]: miss check auth

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[443]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|3|4|75.00%|[115]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|19|22|86.36%|[18, 39, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:844]: I see that the official document does not have the syntax of "| in", so this has not been modified, and other things have been modified. https://docs.starrocks.com/zh-cn/main/sql-reference/sql-statements/data-manipulation/SHOW%20BACKUP

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[456]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|3|4|75.00%|[113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|19|22|86.36%|[18, 39, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|3|4|75.00%|[113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|19|22|86.36%|[18, 38, 51]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 32 / 37 (86.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[468]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|3|4|75.00%|[113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|19|22|86.36%|[18, 38, 51]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by HangyuanLiu** [comment]
please rebase code and I will merge PR as soon as possible

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 41 / 48 (85.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[474]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|19|22|86.36%|[18, 44, 54]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|11|11|100.00%|[]|

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 3]
```diff
@@ -2072,9 +2072,22 @@ show_param ::=
     {:
         RESULT = new ShowExportStmt(db, parser.where, orderByClause, limitClause);
     :}
-    | KW_BACKUP opt_db:db
+    /* Show data statement: used to show data size of specified range */
+    | KW_DATA
+    {:
+        RESULT = new ShowDataStmt(null, null);
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/cup/sql_parser.cup:2078]: rebase meet error ? why add this code?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 79 / 104 (75.96%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[474]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|49|69|71.01%|[327, 346, 351, 357, 373, 374, 382, 389, 398, 404, 409, 410, 415, 417, 418, 419, 422, 423, 431, 437]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|7|8|87.50%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|12|12|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 79 / 104 (75.96%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[474]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|49|69|71.01%|[327, 346, 351, 357, 373, 374, 382, 389, 398, 404, 409, 410, 415, 417, 418, 419, 422, 423, 431, 437]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|7|8|87.50%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|12|12|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 79 / 95 (83.16%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[474]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|49|60|81.67%|[327, 346, 351, 357, 377, 384, 393, 404, 405, 414, 420]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowBackupAnalyzer.java|7|8|87.50%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|12|12|100.00%|[]|

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 3 above)
Comment (melt-code) [reply fe/fe-core/src/main/cup/sql_parser.cup]: already fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 78 / 93 (83.87%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[489]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|66|77|85.71%|[130, 144, 149, 155, 175, 182, 191, 202, 203, 212, 218]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 75 / 90 (83.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[489]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|63|74|85.14%|[121, 135, 140, 146, 166, 173, 182, 193, 194, 203, 209]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 73 / 88 (82.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[489]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|62|73|84.93%|[117, 131, 136, 142, 162, 169, 178, 189, 190, 199, 205]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Review by HangyuanLiu — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 76 / 91 (83.52%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[500]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|65|76|85.53%|[119, 136, 141, 147, 167, 174, 183, 194, 195, 204, 210]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 77 / 92 (83.70%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[500]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|1|4|25.00%|[56, 57, 113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|66|77|85.71%|[119, 136, 141, 147, 168, 175, 184, 195, 196, 205, 211]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 75 / 89 (84.27%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[500]|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|3|4|75.00%|[113]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|62|74|83.78%|[109, 123, 131, 135, 141, 162, 169, 178, 189, 190, 199, 205]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**